add pythia8 automatic configuration on bkg part and other bkg dedicated settings

gconesab opened the pull request.

sawenzel commented on MC/bin/o2dpg_sim_workflow.py:
```diff
@@ -130,14 +132,77 @@ def getDPL_global_options(bigshm=False,nosmallrate=False):
            CONFKEYBKG=' --configKeyValues ' + args.CONFKEYBKG
 
+        # Background PYTHIA configuration
+        BKG_CONFIG_task=createTask(name='genbkgconf')
```
You probably need to add this task to the list of requirements for `BKGtask` further below.

gconesab replied:
do you mean this? workflow['stages'].append(BKG_CONFIG_task) it is below

sawenzel replied:
No. I believe that BKG_CONFIG_task needs to finish before BKGtask can start, right? So you need to add the `needs=[]` field for BKGtask.

gconesab replied:
I tried         BKGtask=createTask(name='bkgsim', lab=["GEANT"], needs=BKG_CONFIG_task['name'], cpu=NWORKERS )
but it does not work

sawenzel replied:
You need a `list`: `needs=[BKG_CONFIG_task['name']]`

gconesab replied:
I think this is what I did on the last commit 
        BKGtask=createTask(name='bkgsim', lab=["GEANT"], needs=[BKG_CONFIG_task['name']], cpu=NWORKERS )

sawenzel replied:
Ok. This was not apparent from these comments. Did you test the whole workflow? Does it run as expected? If yes, I think we can go ahead.

gconesab replied:
yes, it seems to work, although I did not let it end, too hot my computer

sawenzel commented on MC/run/PWGGAJE/run_dirgamma_embedding.sh:
```diff
@@ -55,13 +55,15 @@ echo 'Detector acceptance option ' $PARTICLE_ACCEPTANCE
-                                            -trigger "external" -ini "\$O2DPG_ROOT/MC/config/PWGGAJE/ini/trigger_prompt_gamma.ini"
+${O2DPG_ROOT}/MC/bin/o2dpg_sim_workflow.py -eCM ${CONFIG_ENERGY}                                     \
+                                           -nb ${NBKGEVENTS} --embedding                             \
+                                           -colBkg PbPb -genBkg pythia8 -procBkg "none"              \
```
what does "none" mean as process? Is it different from `inel`?

gconesab replied:
It means no process specified, apparently Pb-Pb Pythia does not need the process

preghenella replied:
@gconesab 
PbPb needs more than just not setting the process.
I suggest that you use process `heavyion` as a label that adds the required settings for pythia8 heavy-ion machinery

gconesab replied:
in mkpy8cfg.py I activate those when Pb-Pb is set, but I agree it is better to specify a HI case, I will do that

sawenzel commented on MC/run/PWGGAJE/run_jets_embedding.sh:
```diff
@@ -15,7 +15,7 @@
 NSIGEVENTS=${NSIGEVENTS:-2}
 NBKGEVENTS=${NBKGEVENTS:-1}
-NTIMEFRAMES=${NTIMEFRAMES:-5}
+NTIMEFRAMES=${NTIMEFRAMES:-1}
```
do you want to commit this change?

gconesab replied:
It is an example I was testing, I will change it back

gconesab commented:
Ciao @sawenzel and @preghenella, I think I addressed all your comments, thanks!
I added a new process "heavy_ion" and renamed the one called "heavy" to "heavy_q". Let me know if you agree

gconesab commented:
Hi guys, any extra comment? If not, I will merge later today.
